timeline/templatetags/timeline_tags.py

```diff
@@ -86,8 +86,15 @@ def get_app_icon_html(event, plugin_lookup):
         if plugin_name in plugin_lookup.keys():
             plugin = plugin_lookup[plugin_name]
             entry_point = getattr(plugin, 'entry_point_url_id', None)
-            if entry_point:
-                url = reverse(entry_point, kwargs=url_kwargs)
+            if entry_point in plugin_lookup.items:
```

Comment:
This try-except is good as it fixes the immediate problem of the crash.

We still need to have the check for the plugin type, as specified in the issue. If the plugin is a `SiteAppPlugin`, we should not provide `project` in `url_kwargs`, as site apps will not have that in their entry point. Please add that.